src/codeu/chat/client/View.java

```diff
@@ -40,24 +47,169 @@
 
   private final static Logger.Log LOG = Logger.newLog(View.class);
 
-  private final ConnectionSource source;
+  private final Connection source;
+  
+  private ClientConnection client;
+  
+  // Pauses the main thread to allow ClientConnection 
+  // to parse the newly received data.
+  public CountDownLatch latch = new CountDownLatch(1);
+  
+  // Post-processing Variables for Client and Server
+  private User signedInUser = null;
+  // getUsersExcluding()
+  private Collection<User> users = new ArrayList<>();
+  // getAllConversations()
+  private Collection<ConversationSummary> summaries;
+  // getMessages()
+  private Collection<Message> messages;
+  // getMessageById
+  private Message getMessageById;
+  // getMessageCount
+  int messageCount = 0;
+  //  getConversations
+  Collection<Conversation> conversations = new ArrayList<>();
+  
+  // FROM CONTROLLER
+  // newUser
+  private User newUser;
+  private Conversation newConversation;
+  private Message newMessage;
+  
+  // Reference to ClientContext
+  public ClientContext clientContext;
+  
+  // Reference to the GUI
+  public MainChatPage mainChatPage;
+  
+
+  /**
+   * ClientConnections manages the clients connection on the client side
+   * by checking for an incoming information and parsing it accordingly.
+   * @author malikg
+   *
+   */
+  class ClientConnection extends Thread {
```

Comment:
[insert Google's favorite question]. So are you gonna work on implementing a better way rather than having a timer?